src/nm-dispatcher.c

```diff
@@ -15,7 +15,7 @@
  * with this program; if not, write to the Free Software Foundation, Inc.,
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  *
- * Copyright (C) 2004 - 2017 Red Hat, Inc.
+ * Copyright (C) 2004 - 2018 Red Hat, Inc.
```

Comment:
Commit-message: "Follow up patches make dispatcher optional."

the dispatcher was always optional, you were always welcome to mask the service. The commit message makes it sound that wouldn't be the case.

Reply:
The claim that the fast-path is not hit in usual scenarios does not seem correct. The monitors are also for `pre-up.d` and `pre-down.d` directories, and these are quite often empty. Especially `pre-down.d` is always empty, and AFAIK, only `NetworkManager-dispatcher-routing-rules` package installs a script in `pre-up.d` (and this is an optional package, only on RHEL/Fedora).

The patch saves 150 LOC. Sure, and extra D-Bus call just to figure out that there is nothing to do is relatively fast. I say, "relatively", because NM is competing with networkd, which can configure an interface in a matter of milliseconds. How long does it really take? You need to context-switch to dbus-daemon, NM-dispatcher, dbus-daemon and back.

Reply:
> Commit-message: "Follow up patches make dispatcher optional."
> 
> the dispatcher was always optional, you were always welcome to mask the service. The commit message makes it sound that wouldn't be the case.

Okay. Dropped that part.

> The claim that the fast-path is not hit in usual scenarios does not seem correct. The monitors are also for `pre-up.d` and `pre-down.d` directories, and these are quite often empty. Especially `pre-down.d` is always empty, and AFAIK, only `NetworkManager-dispatcher-routing-rules` package installs a script in `pre-up.d` (and this is an optional package, only on RHEL/Fedora).
> 
> The patch saves 150 LOC. Sure, and extra D-Bus call just to figure out that there is nothing to do is relatively fast. I say, "relatively", because NM is competing with networkd, which can configure an interface in a matter of milliseconds. How long does it really take? You need to context-switch to dbus-daemon, NM-dispatcher, dbus-daemon and back.

Ah, I didn't really think of that. You're right.

A quick look at the Bustle graph shows 12ms first (activation i suppose) and then under a milisecond for subsequent calls. I guess that's okay?
![lala](https://user-images.githubusercontent.com/92695/49239515-3cfb9500-f403-11e8-95c4-f20d9e994b43.png)

Reply:
> I don't think we need to optimize this further for simplicity, because IMO it's self-contained and simple already.

I agree, the complexity of monitoring dispatcher files doesn't seem too much to me, and offers some (small) performance improvements.

Reply:
This is an important part though:

> Let's remove them now, making it easier to modify the dispatcher to look for scripts in other places.

We'd need more watches and we'd also need the daemon to know about them, which adds complexity to the daemon I'd like to avoid.